lib/experimental/class-theme-json-webfonts-handler.php

```diff
@@ -0,0 +1,537 @@
+					foreach ( $font_families as $font_family ) {
+
+						// Skip if fontFace is not defined.
+						if ( empty( $font_family['fontFace'] ) ) {
+							continue;
+						}
+
+						// Skip if fontFace is not an array of webfonts.
+						if ( ! is_array( $font_family['fontFace'] ) ) {
+							continue;
+						}
+
+						$webfonts = array_merge( $webfonts, $font_family['fontFace'] );
+					}
+				}
+
+				return $webfonts;
+			}
+
+			/**
+			 * Transform each `src` into an URI by replacing the `file:./`
+			 * placeholder from theme.json.
+			 *
+			 * The absolute path to the webfont file(s) Cannot be defined in
+			 * theme.json. `file:./` is the placeholder which is replaced by
+			 * the theme's URL path to the theme's root.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @param array $src Webfont file(s) `src`.
+			 *
+			 * @return array Webfont's `src` in URI.
+			 */
+			private function transform_src_into_uri( array $src ) {
+				foreach ( $src as $key => $url ) {
+					// Tweak the URL to be relative to the theme root.
+					if ( ! str_starts_with( $url, 'file:./' ) ) {
+						continue;
+					}
+
+					$src[ $key ] = get_theme_file_uri( str_replace( 'file:./', '', $url ) );
+				}
+
+				return $src;
+			}
+
+			/**
+			 * Converts the face-face properties (i.e. keys) into kebab-case.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @param array $font_face Font face to convert.
+			 *
+			 * @return array Font faces with each property in kebab-case format.
+			 */
+			private function convert_keys_to_kebab_case( array $font_face ) {
+				foreach ( $font_face as $property => $value ) {
+					$kebab_case               = _wp_to_kebab_case( $property );
+					$font_face[ $kebab_case ] = $value;
+					if ( $kebab_case !== $property ) {
+						unset( $font_face[ $property ] );
+					}
+				}
+
+				return $font_face;
+			}
+
+			/**
+			 * Validate a webfont.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @param array $webfont The webfont arguments.
+			 *
+			 * @return array|false The validated webfont arguments, or false if the webfont is invalid.
+			 */
+			public function validate_webfont( $webfont ) {
+				$webfont = wp_parse_args(
+					$webfont,
+					array(
+						'font-family'  => '',
+						'font-style'   => 'normal',
+						'font-weight'  => '400',
+						'font-display' => 'fallback',
+						'src'          => array(),
+					)
+				);
+
+				// Check the font-family.
+				if ( empty( $webfont['font-family'] ) || ! is_string( $webfont['font-family'] ) ) {
+					trigger_error( __( 'Webfont font family must be a non-empty string.', 'gutenberg' ) );
+
+					return false;
+				}
+
+				// Make a `src` property is defined.
+				if ( empty( $webfont['src'] ) || ( ! is_string( $webfont['src'] ) && ! is_array( $webfont['src'] ) ) ) {
+					trigger_error( __( 'Webfont src must be a non-empty string or an array of strings.', 'gutenberg' ) );
+
+					return false;
+				}
+
+				// Validate the 'src' property.
```

Comment:
```suggestion
				// Validate the `src` property.
```

For consistency with the block above.